File: Dataset/PalindromeChecker.java
public aspect LoggingAspect {
    pointcut methodCalls(): execution(* *(..));
    before(): methodCalls() {
        System.out.println("Method called: " + thisJoinPoint.getSignature());
    }
}
public class PalindromeChecker {
    public boolean isPalindrome(int number) {
        int original = number;
        int reversed = 0;
        while (number != 0) {
            int digit = number % 10;
            reversed = reversed * 10 + digit;
            number /= 10;
        }
        return original == reversed;
    }
    public static void main(String[] args) {
        PalindromeChecker checker = new PalindromeChecker();
        int number = 121;
        boolean result = checker.isPalindrome(number);
        System.out.println("Is " + number + " a palindrome? " + result);
    }
}

File: Dataset/PalindromeChecker.rb
module LoggingAspect
    def is_palindrome(number)
      puts "Method called: is_palindrome"
      super
    end
  end
  class PalindromeChecker
    prepend LoggingAspect
    def is_palindrome(number)
      original = number
      reversed = 0
      while number != 0
        digit = number % 10
        reversed = reversed * 10 + digit
        number /= 10
      end
      original == reversed
    end
  end
  checker = PalindromeChecker.new
  number = 121
  result = checker.is_palindrome(number)
  puts "Is #{number} a palindrome? #{result}"
  